creed/tiling_utils_test_cases.py

```diff
@@ -0,0 +1,214 @@
+        return 'N', True, 3, [3, 7, 11, 15, 19], 10
+
+    def case_no_region_interval_even_partition(self):
+        return 'N', True, 4, [4, 13], 9
+
+    def case_region_no_interval(self):
+        return 'Y', False, 0, [8], 9
+
+    def case_region_interval_uneven_partition(self):
+        return 'Y', True, 3, [3, 7, 10, 12, 16, 20], 11
+
+    def case_region_interval_even_partition(self):
+        return 'Y', True, 4, [4, 9, 14], 10
+
+
+# TMA rhombus coordinate validation
+class ValidateRhombusCoordsTests:
+    def case_top_left_failure(self):
+        top_left = XYCoord(100, 200)
+        top_right = XYCoord(50, 100)
+        bottom_left = XYCoord(150, 300)
+        bottom_right = XYCoord(100, 200)
+
+        return top_left, top_right, bottom_left, bottom_right
+
+    def case_bottom_left_failure(self):
+        top_left = XYCoord(100, 200)
+        top_right = XYCoord(150, 100)
+        bottom_left = XYCoord(150, 300)
+        bottom_right = XYCoord(100, 200)
+
+        return top_left, top_right, bottom_left, bottom_right
+
+    def case_top_right_failure(self):
+        top_left = XYCoord(100, 200)
+        top_right = XYCoord(150, 100)
+        bottom_left = XYCoord(150, 300)
+        bottom_right = XYCoord(200, 200)
+
+        return top_left, top_right, bottom_left, bottom_right
+
+    def case_bottom_right_failure(self):
+        top_left = XYCoord(100, 400)
+        top_right = XYCoord(150, 100)
+        bottom_left = XYCoord(150, 300)
+        bottom_right = XYCoord(200, 200)
+
+        return top_left, top_right, bottom_left, bottom_right
+
+    def case_success(self):
+        top_left = XYCoord(100, 400)
+        top_right = XYCoord(150, 300)
+        bottom_left = XYCoord(150, 300)
+        bottom_right = XYCoord(200, 200)
+
+        return top_left, top_right, bottom_left, bottom_right
+
+
+# TMA rhombus inputs for coordinate generation and TMA FOV list generation
+_TMA_RHOMBUS_X_COORDS = (1500, 1666, 1833, 2000)
+
+
+class RhombusCoordInputTests:
+    def case_rectangle(self):
+        coords = [
+            XYCoord(1500, 2000),
+            XYCoord(2000, 2000),
+            XYCoord(1500, 1000),
+            XYCoord(2000, 1000)
+        ]
+
+        # since this is a simple rectangle we can easily and programmatically define these coords
+        actual_pairs = []
+        for x_coord in _TMA_RHOMBUS_X_COORDS:
+            actual_pairs.extend([
+                (x_coord, y_coord)
+                for y_coord in reversed(np.arange(1000, 2500, 500))
+            ])
+
+        return coords, actual_pairs
+
+    def case_x_slant(self):
+        coords = [
+            XYCoord(1500, 2000),
+            XYCoord(2000, 2000),
+            XYCoord(1600, 1000),
+            XYCoord(2200, 1000)
+        ]
+
+        # total x-offset is 150 (average of the left-offset (100) and right-offset (200))
+        # 3 fovs along the y-axis, so divide by 2 to get individual fov offset (75)
```

Comment:
This comment is useful, but it would be better to define the x_increment using the relationship you describe, so that if coords changes this value changes automatically too